src/components/Navbar/Navbar.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useCallback, useState } from "react"
+import { Toolbar } from "@mui/material"
+import { MainMenu } from "./MainMenu/MainMenu"
+import {
+  i18nChainSelectorDefault,
+  navbarMainTitlesI18N as i18nNavbarTitlesDefault,
+  navbarSubmenu,
+} from "./Navbar.defaults"
+import { SubMenu } from "./SubMenu/SubMenu"
+import { ChainSelector } from "../ChainSelector"
+import { Logo } from "../Logo/Logo"
+import { TabletAndBelow, useTabletAndBelowMediaQuery } from "../Media"
+import { UserMenu } from "../UserMenu"
+import { i18n as i18nUserMenuDefault } from "../UserMenu/UserMenu.i18n"
+import { NavbarPages, NavbarProps } from "./Navbar.types"
+import {
+  AppBarDesktopWrapper,
+  AppBarRightWrapper,
+  AppBarWrapper,
+  DclAppBar,
+  LogoLink,
+  MenuIcon,
+  MenuIconBar,
+  MenuModal,
+} from "./Navbar.styled"
+
+const Navbar = React.memo((props: NavbarProps) => {
+  const {
+    activePage,
+    chains,
+    chainBeingConfirmed,
+    onSelectChain,
+    selectedChain,
+    isSignedIn,
+    i18nNavbar = i18nNavbarTitlesDefault,
+    i18nUserMenu = i18nUserMenuDefault,
+    i18nChainSelector = i18nChainSelectorDefault,
+    submenuItems = navbarSubmenu,
+    onClickNavbarItem,
+    ...userMenuProps
+  } = props
+  const [toggle, setToggle] = useState(false)
+  const [selectedMenu, setSelectedMenu] = useState<NavbarPages | boolean>()
+
+  const [menuMobileOpen, setMenuMobileOpen] = useState(false)
+
+  const isTabletAndBelow = useTabletAndBelowMediaQuery()
+
+  const handleToggle = useCallback(
+    (
+      _e: React.MouseEvent<HTMLElement>,
+      show: boolean,
+      section: NavbarPages
+    ) => {
+      setToggle(show)
+      setSelectedMenu(show && section)
+    },
+    [setToggle, setSelectedMenu]
+  )
+
+  const handleClickMenu = useCallback(
+    (
+      event: React.MouseEvent<HTMLElement, MouseEvent>,
+      options: { eventTrackingName: string; url?: string; isExternal?: boolean }
+    ) => {
+      onClickNavbarItem && onClickNavbarItem(event, options)
+    },
+    [onClickNavbarItem]
+  )
+
+  const handleMobileToggle = useCallback(
+    (_e: React.MouseEvent<HTMLElement>, show: boolean) => {
+      if (!show) {
+        setToggle(false)
+      }
+      setMenuMobileOpen(show)
+    },
+    [setToggle, setMenuMobileOpen]
+  )
+
+  const handleUserMenuOpen = (
```

Comment:
This function is missing the use of `useCallback` to define it:
```suggestion
  const handleUserMenuOpen = useCallback((
```
Would you mind adding it?